### ISSUE TITLE
2D alpha shapes with periodic Delaunay triangulation

### ISSUE THREAD

**Issue by grzjab** [CGAL/cgal#1258]
Can you add  OUTSIDE_CONVEX_HULL and  OUTSIDE_AFFINE_HULL in  enum Locate_type in Periodic_2_triangulation_2.h for compatibility with Alpha_shape_2?
Periodic_3_triangulation_3.h has it:

```
 OUTSIDE_CONVEX_HULL, // unused, for compatibility with Alpha_shape_3
 OUTSIDE_AFFINE_HULL }; // unused, for compatibility with Alpha_shape_3
```

I found the bug while trying to compile the following code:

```
#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
#include <CGAL/Periodic_2_triangulation_filtered_traits_2.h>
#include <CGAL/Periodic_2_Delaunay_triangulation_2.h>
#include <CGAL/Triangulation_vertex_base_with_info_2.h>
#include <CGAL/Alpha_shape_2.h>
 //**********TYPEDEFS********************
 typedef CGAL::Exact_predicates_exact_constructions_kernel K;
 typedef CGAL::Periodic_2_triangulation_traits_2<K> GT;
 typedef CGAL::Triangulation_vertex_base_with_info_2<unsigned, K> Vb_info;
 typedef CGAL::Alpha_shape_vertex_base_2<K, Vb_info>    Vb;
 typedef CGAL::Alpha_shape_face_base_2<K>                                               Fb;
 typedef CGAL::Triangulation_data_structure_2<Vb, Fb>    Tds;
 typedef CGAL::Periodic_2_Delaunay_triangulation_2<GT, Tds>  PDT;
 typedef CGAL::Alpha_shape_2<PDT> Alpha_shape_2;

 int main()
 {
        Alpha_shape_2 as;
 }
```

EDIT: Output from compiler:

```
In file included from /home/grzegorz/work/delaunay/example.cpp:5:0:
 /home/grzegorz/work/CGAL-4.8/include/CGAL/Alpha_shape_2.h: In instantiation of ‘class CGAL::Alpha_shape_2<CGAL::Periodic_2_Delaunay_triangulation_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epeck>, CGAL::Triangulation_data_structure_2<CGAL::Alpha_shape_vertex_base_2<CGAL::Epeck, CGAL::Triangulation_vertex_base_with_info_2<unsigned int, CGAL::Epeck> >, CGAL::Alpha_shape_face_base_2<CGAL::Epeck> > > >’:
 /home/grzegorz/work/delaunay/example.cpp:19:16:   required from here
 /home/grzegorz/work/CGAL-4.8/include/CGAL/Alpha_shape_2.h:107:13: error: no members matching ‘CGAL::Periodic_2_Delaunay_triangulation_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epeck>, CGAL::Triangulation_data_structure_2<CGAL::Alpha_shape_vertex_base_2<CGAL::Epeck, CGAL::Triangulation_vertex_base_with_info_2<unsigned int, CGAL::Epeck> >, CGAL::Alpha_shape_face_base_2<CGAL::Epeck> > >::OUTSIDE_CONVEX_HULL’ in ‘class CGAL::Periodic_2_Delaunay_triangulation_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epeck>, CGAL::Triangulation_data_structure_2<CGAL::Alpha_shape_vertex_base_2<CGAL::Epeck, CGAL::Triangulation_vertex_base_with_info_2<unsigned int, CGAL::Epeck> >, CGAL::Alpha_shape_face_base_2<CGAL::Epeck> > >’
    using Dt::OUTSIDE_CONVEX_HULL;
              ^
 /home/grzegorz/work/CGAL-4.8/include/CGAL/Alpha_shape_2.h:108:13: error: no members matching ‘CGAL::Periodic_2_Delaunay_triangulation_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epeck>, CGAL::Triangulation_data_structure_2<CGAL::Alpha_shape_vertex_base_2<CGAL::Epeck, CGAL::Triangulation_vertex_base_with_info_2<unsigned int, CGAL::Epeck> >, CGAL::Alpha_shape_face_base_2<CGAL::Epeck> > >::OUTSIDE_AFFINE_HULL’ in ‘class CGAL::Periodic_2_Delaunay_triangulation_2<CGAL::Periodic_2_triangulation_traits_2<CGAL::Epeck>, CGAL::Triangulation_data_structure_2<CGAL::Alpha_shape_vertex_base_2<CGAL::Epeck, CGAL::Triangulation_vertex_base_with_info_2<unsigned int, CGAL::Epeck> >, CGAL::Alpha_shape_face_base_2<CGAL::Epeck> > >’
    using Dt::OUTSIDE_AFFINE_HULL;
```

**Comment by mglisse** [comment]
Some links for reference:
https://github.com/CGAL/cgal-public-dev/tree/Periodic_2_triangulation_2-alpha-shapes-nkruithof
https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Periodic_alpha_2

**Comment by sloriot** [comment]
@MaelRL do you think that your recent changes might solves the remaining issues to get 2D alpha shapes working with 2D periodic triangulations? (maybe with some commit from @nicokruithof branch)

**Comment by MaelRL** [comment]
What were the remaining issues with this the P2T2-alpha_shapes branch? The discussion on the wiki page of the feature does not explain why it stalled.

**Comment by sloriot** [comment]
From what I remember there was an issue with the `lazy_alpha` that you probably have solved. I guess we need to give it a try to see if there is anything left that breaks as @nicokruithof might also not remember the issues.

**Comment by lrineau** [comment]
@grzjab This issue has been fixed by the merge of the pull-request #2369 in the `master` branch of CGAL, and the fix will be in CGAL-4.12, planned 2018 Q1.
